runtime/js/99_main.js

```diff
@@ -74,66 +74,77 @@ delete Object.prototype.__proto__;
   const onerror = () => {};
 
   function postMessage(data) {
-    const dataJson = JSON.stringify(data);
-    const dataIntArray = encoder.encode(dataJson);
+    const dataIntArray = core.serialize(data);
     opPostMessage(dataIntArray);
   }
 
   let isClosing = false;
-  async function workerMessageRecvCallback(data) {
-    const msgEvent = new MessageEvent("message", {
-      cancelable: false,
-      data,
-    });
-
-    try {
-      if (globalThis["onmessage"]) {
-        const result = globalThis.onmessage(msgEvent);
-        if (result && "then" in result) {
-          await result;
-        }
-      }
-      globalThis.dispatchEvent(msgEvent);
-    } catch (e) {
-      let handled = false;
-
-      const errorEvent = new ErrorEvent("error", {
-        cancelable: true,
-        message: e.message,
-        lineno: e.lineNumber ? e.lineNumber + 1 : undefined,
-        colno: e.columnNumber ? e.columnNumber + 1 : undefined,
-        filename: e.fileName,
-        error: null,
+  async function getHostMessages() {
```

Comment:
What's the advantage of polling for messages from within runtime instead of explicitly calling a function like `workerMessageRecvCallback`?

Also, please rename to `pollForMessages`.

Reply:
I renamed the function to `pollForMessages`.
I think its simpler to just register an ops function, also this is how the host receives messages from the workers too.